src/providers/krb5/krb5_child.c

```diff
@@ -1683,20 +1692,48 @@ static krb5_error_code validate_tgt(struct krb5_req *kr)
 
     /* Try to find and send the PAC to the PAC responder.
      * Failures are not critical. */
-    if (kr->send_pac) {
+    if (kr->send_pac || kr->cli_opts->check_pac != NULL) {
         kerr = sss_extract_pac(kr->ctx, validation_ccache, validation_princ,
-                               kr->creds->client, keytab, &pac_authdata);
+                               kr->creds->client, keytab,
+                               kr->cli_opts->check_pac_flags, &pac_authdata);
         if (kerr != 0) {
+            if (kerr == ERR_CHECK_PAC_FAILED) {
+                DEBUG(SSSDBG_CRIT_FAILURE,
+                      "PAC check failed for principal [%s].\n", kr->name);
+                goto done;
+            }
             DEBUG(SSSDBG_OP_FAILURE, "sss_extract_and_send_pac failed, group " \
                                       "membership for user with principal [%s] " \
                                       "might not be correct.\n", kr->name);
             kerr = 0;
             goto done;
         }
+    }
+
+    if (kr->send_pac) {
+        if(unsetenv("_SSS_LOOPS") != 0) {
+            DEBUG(SSSDBG_CRIT_FAILURE, "Failed to unset _SSS_LOOPS, "
+                      "sss_pac_make_request will most certainly fail.\n");
+        }
 
         kerr = sss_send_pac(pac_authdata);
-        krb5_free_authdata(kr->ctx, pac_authdata);
+
+        if(setenv("_SSS_LOOPS", "NO", 0) != 0) {
+            DEBUG(SSSDBG_OP_FAILURE, "Failed to set _SSS_LOOPS.\n");
+        }
+
         if (kerr != 0) {
+            if (kerr == ERR_CHECK_PAC_FAILED) {
+                DEBUG(SSSDBG_CRIT_FAILURE,
+                      "PAC for principal [%s] is not valid.\n", kr->name);
+                goto done;
+            }
+            if (kr->cli_opts->check_pac != NULL) {
+                DEBUG(SSSDBG_CRIT_FAILURE,
```

Comment:
Is backtrace useful here or would `SSSDBG_IMPORTANT_INFO` be enough?

Reply:
fixed